### ISSUE TITLE
Really use lora?

### ISSUE THREAD

**Issue by Maybeetw** [alexdenker/SteerableConditionalDiffusion#1]
When I was running the code, I found that these, 

in adapt/adaptation.py ，lines 78
```
elif method == 'lora':
        """ 
        Implement LoRA: https://arxiv.org/pdf/2106.09685.pdf 

        Adding LoRA modules to nn.Conv1d, nn.Conv2d (should we also add to nn.Linear?)
         + retraining all biases (only a negligible number of parameters)
        """
        score.requires_grad_(False)

        # should we also do something to the last layer?
        for name, param in score.named_parameters():
            #print(name)
            if "bias" in name and not "emb_layers" in name:
                param.requires_grad = True
        
        #print(score.out)
        
        inject_trainable_lora_extended(score, r=r) 

        new_num_params = sum([p.numel() for p in score.parameters()])
        trainable_params = sum([p.numel() for p in score.parameters() if p.requires_grad])
        print(f'% of trainable params: {trainable_params/new_num_params*100}')

        score.requires_grad_(True)
```
Why set all parameter gradients to True
if used, would become full parameter fine-tuning and still belong to the Lora category. 

But commenting out this line will result in an error,Causing inability to train

Excuse me, did you encounter any version issues while uploading the code

**Comment by alexdenker** [comment]
Thank you for pointing out this bug! Of course the line `score.requires_grad_(True)` has to be removed. 

We used a modified version of the AttentionUNet (https://github.com/openai/guided-diffusion/blob/main/guided_diffusion/nn.py) where I removed the checkpointing, ema and fp16 as it was not used in our experiments. For the release of this code, I put the original guided_diffusion code in and it turns out that the `CheckpointFunction` (line 142 https://github.com/openai/guided-diffusion/blob/main/guided_diffusion/nn.py) does not work with LoRA, see also this thread: https://discuss.pytorch.org/t/get-runtimeerror-one-of-the-differentiated-tensors-does-not-require-grad-in-pytorch-lightning/179738

I pushed a fix (see commit `46164e0`).

Does the example code
`python conditional_sampling.py --train_on=ellipses --test_on=walnut --method=scd --K=16 --r=8   --lr=1e-4  --gamma=0.1  --skip=20 --num_angles=60 --noise_std=0.05 --alphatv=1e-3`
now work? 

Thanks!

**Comment by Maybeetw** [comment]
It's already working now. Thank you very much for your response quickly。
